### PR TITLE
feat: Add 25Mar2020 Virtual Session fields for non-offline sessions

### PR DESCRIPTION
**Pull request by lukehesluke**
Add Virtual Session fields that were solidified in the 25th March 2020 OA call

This includes all the fields defined in the [https://developer.openactive.io/publishing-data/virtual-events](Virtual Events Guidance) except `beta:donationPaymentUrl`

## QA

Example of a golden session:

Screenshot 2020-03-27 at 20 53 31

Online sessions have no `location`:

Screenshot 2020-03-27 at 20 58 00

Example of an affiliated location:

Screenshot 2020-03-27 at 20 59 09

### CODE REVIEW

File: data/itemGenerator.js [hunk 1]
```diff
@@ -119,7 +124,7 @@ function generateOffer(age, baseUrl, modified, golden, free) {
 
 function generateImages(golden) {
   var out = [];
-  var imageCount = faker.random.number(golden ? {min: 4, max: 6} : 6);
+  var imageCount = faker.random.number(golden ? {min: 4, max: 6} : { max: 6 });
```
Comment (lukehesluke) [inline data/itemGenerator.js:127]: My error checker gets confused when function call signatures are mixed with a single call (i.e. `number(max: number)` vs `number({ max: number, min: number })`).

So I've updated it to both use the same signature (with the `min`/`max` object)

File: data/itemGenerator.js [hunk 2]
```diff
@@ -354,21 +376,45 @@ function generateSlot(startDateString, durationMinutes, tzid, modified, maximumU
   return {data, subId: startDate.unix(), deleted: false };
 }
 
-function generateSubEvent(startDateString, duration, tzid, modified, maximumAttendeeCapacity, baseUrl, golden) {
-  var startDate = moment.tz(startDateString, tzid);
-  var endDate = startDate.clone().add(moment.duration(duration));
+/**
+ * @param {Date} startDate
```
Comment (lukehesluke) [inline data/itemGenerator.js:380]: This was not a string - it was a `Date`

File: data/itemGenerator.js [hunk 3]
```diff
@@ -421,6 +467,82 @@ function generatePerson(baseUrl, golden) {
...
+      {
+        "type": "OpeningHoursSpecification",
+        "dayOfWeek": "https://schema.org/Saturday",
+        "opens": "07:15",
+        "closes": "17:30"
+      }
+    ]
+  };
+}
+
```
Comment (lukehesluke) [inline data/itemGenerator.js:545]: This is an exact copy of the `Place` object that was created in `generateSessionSeriesItemData`. I have made no updates to it other than refactoring it into its own function

**Review by nickevansuk — APPROVED** [review]

**Comment by nickevansuk** [comment]
Looks great, thanks for the contribution @lukehesluke !

**Pull request merged**